lxd/patches.go

```diff
@@ -950,6 +951,115 @@ func patchStorageRenameCustomISOBlockVolumes(name string, d *Daemon) error {
+		if api.StatusErrorCheck(err, http.StatusNotFound) {
+			return nil
+		}
+
+		return fmt.Errorf("Failed getting storage pool names: %w", err)
+	}
+
+	volTypeCustom := dbCluster.StoragePoolVolumeTypeCustom
+	customPoolVolumes := make(map[string][]*db.StorageVolume, 0)
+
+	err = s.DB.Cluster.Transaction(s.ShutdownCtx, func(ctx context.Context, tx *db.ClusterTx) error {
+		for _, pool := range pools {
+			// Get storage pool ID.
+			poolID, err := tx.GetStoragePoolID(ctx, pool)
+			if err != nil {
+				return fmt.Errorf("Failed getting storage pool ID of pool %q: %w", pool, err)
+			}
+
+			// Get the pool's custom storage volumes.
+			customVolumes, err := tx.GetStoragePoolVolumes(ctx, poolID, false, db.StorageVolumeFilter{Type: &volTypeCustom})
+			if err != nil {
+				return fmt.Errorf("Failed getting custom storage volumes of pool %q: %w", pool, err)
+			}
+
+			if customPoolVolumes[pool] == nil {
+				customPoolVolumes[pool] = []*db.StorageVolume{}
+			}
+
+			customPoolVolumes[pool] = append(customPoolVolumes[pool], customVolumes...)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	isSelectedPatchMember, err := selectedPatchClusterMember(d)
+	if err != nil {
+		return err
+	}
+
+	for poolName, volumes := range customPoolVolumes {
+		// Load storage pool.
+		p, err := storagePools.LoadByName(s, poolName)
+		if err != nil {
+			return fmt.Errorf("Failed loading pool %q: %w", poolName, err)
+		}
+
+		// Ensure the renaming is done only on the selected patch cluster member for remote storage pools.
+		if p.Driver().Info().Remote && !isSelectedPatchMember {
+			continue
+		}
+
+		for _, vol := range volumes {
+			// In a non-clusted environment ServerName will be empty.
+			if s.ServerName != "" && vol.Location != s.ServerName {
+				continue
+			}
+
+			// Exclude non-ISO custom volumes.
+			if vol.ContentType != dbCluster.StoragePoolVolumeContentTypeNameISO {
+				continue
+			}
+
+			// The existing volume using the actual *.iso suffix has ContentTypeISO.
+			existingVol := storageDrivers.NewVolume(p.Driver(), p.Name(), storageDrivers.VolumeTypeCustom, storageDrivers.ContentTypeISO, project.StorageVolume(vol.Project, vol.Name), nil, nil)
+
+			hasVol, err := p.Driver().HasVolume(existingVol)
+			if err != nil {
+				return err
+			}
+
+			// patchStorageRenameCustomISOBlockVolumes might have already set the *.iso suffix.
+			// Check if the storage volume isn't already renamed.
+			if hasVol {
+				continue
+			}
+
+			// We need to use ContentTypeBlock here in order for the driver to figure out the correct (old) location.
+			oldVol := storageDrivers.NewVolume(p.Driver(), p.Name(), storageDrivers.VolumeTypeCustom, storageDrivers.ContentTypeBlock, project.StorageVolume(vol.Project, vol.Name), nil, nil)
+
+			err = p.Driver().RenameVolume(oldVol, fmt.Sprintf("%s.iso", oldVol.Name()), nil)
+			if err != nil {
+				return fmt.Errorf("Failed renaming volume: %w", err)
```

Comment:
Would be useful to include the volume info here into the error in case it fails.

Reply:
Fixed.